master/static/migrations/20240409101349_add-arbitrary-metadata-table.tx.up.sql

```diff
@@ -0,0 +1,32 @@
+-- ALTER RUNS TABLE
+ALTER TABLE runs ADD COLUMN metadata JSONB;
+-- CREATE INDEX TABLE
+CREATE TABLE runs_metadata_index (
+    id SERIAL PRIMARY KEY,
+    run_id INTEGER,
+    flat_key VARCHAR,
+    value TEXT,
```

Comment:
For filtering on arbitrary metadata will we have to cast the type for the value when doing >, =, >=, etc.. operations? There may be some performance concerns over this but I'm not quite sure.

Reply:
Yep, this will ensure if we're doing equality & inequality checks are still efficient, when we start to filtering for ranges or < & >, the we'll need to need to perform casting.  If we see performance declines from this we can do the sparse row approach with columns dedicated to each type.